src/DetectSecretsLinter.php

```diff
@@ -49,47 +49,36 @@ public function getInstallInstructions() {
   }
 
   protected function getMandatoryFlags() {
-      return array('-v', 'scan');
+      return array('scan');
   }
 
   protected function parseLinterOutput($path, $err, $stdout, $stderr) {
 
-      $lines = phutil_split_lines($stdout, false);
+      $json = json_decode($stdout, true);
+      $json_results = $json["results"];
 
       $messages = array();
-      $output_string = array();
-
-      foreach($lines as $result) {
-	 if ($result !== '') {
-             if (preg_match("/\bresults\b/i", $result)) {
-                 $initial = array_search($result, $lines);
-             }
-
-             if (preg_match("/\bversion\b/i", $result)) {
-                 $final = array_search($result, $lines);
-	     }
-	 }
-      }
-
-      for($i = $initial+1; $i < $final; $i += 1) {
-          $output_string[] = $lines[$i];    
-      }
 
       $error_message = "Looks like you are about to commit secrets to this repo. Please avoid this practice\n\n";
       $error_message .= "Possible mitigations:\n";
       $error_message .= "1. For information about putting your secrets in a safer place, please refer pinch/knox \n";
       $error_message .= "2. If secret has already been committed please rotate that secret. If rotation is taking significant time then please contact #security_related slack channel\n";
       $error_message .= "3. If its a test file with secrets (not belonging to any prod service) mark false positives with an inline `pragma: allowlist secret` comment\n";
```

Comment:
Some of this message text is Pinterest-specific. Because this is an open source project, we should make this text generic or configurable.